From da08507a82fb2dd124a9ef06bd5157dfecb9e99f Mon Sep 17 00:00:00 2001
From: John Howard <howardjohn@google.com>
Date: Thu, 18 Nov 2021 10:49:48 -0800
Subject: Fix quoting issue on environment tests (#36141)

* Fix quoting issue on environment tests

* fix select flag as well
---
 tests/integration/tests.mk | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tests/integration/tests.mk b/tests/integration/tests.mk
index 7499c70259..1b2527f3f7 100644
--- a/tests/integration/tests.mk
+++ b/tests/integration/tests.mk
@@ -114,7 +114,7 @@ test.integration.kube.presubmit: | $(JUNIT_REPORT) check-go-tag
 .PHONY: test.integration.kube.reachability
 test.integration.kube.reachability: | $(JUNIT_REPORT) check-go-tag
 	$(GO) test -p 1 ${T} -tags=integ ./tests/integration/security/ -timeout 30m \
-	${_INTEGRATION_TEST_FLAGS} \
+	${_INTEGRATION_TEST_FLAGS} ${_INTEGRATION_TEST_SELECT_FLAGS} \
 	--test.run=TestReachability \
 	2>&1 | tee >($(JUNIT_REPORT) > $(JUNIT_OUT))
 
@@ -124,11 +124,11 @@ test.integration.kube.reachability: | $(JUNIT_REPORT) check-go-tag
 test.integration.kube.environment: | $(JUNIT_REPORT) check-go-tag
 ifeq (${JOB_TYPE},postsubmit)
 	PATH=${PATH}:${ISTIO_OUT} $(GO) test -p 1 ${T} -tags=integ ./tests/integration/... -timeout 30m \
-	${_INTEGRATION_TEST_FLAGS} \
+	${_INTEGRATION_TEST_FLAGS} ${_INTEGRATION_TEST_SELECT_FLAGS} \
 	2>&1 | tee >($(JUNIT_REPORT) > $(JUNIT_OUT))
 else
 	PATH=${PATH}:${ISTIO_OUT} $(GO) test -p 1 ${T} -tags=integ ./tests/integration/security/ ./tests/integration/pilot -timeout 30m \
-	${_INTEGRATION_TEST_FLAGS} \
-	--test.run=TestReachability|TestTraffic \
+	${_INTEGRATION_TEST_FLAGS} ${_INTEGRATION_TEST_SELECT_FLAGS} \
+	--test.run="TestReachability|TestTraffic" \
 	2>&1 | tee >($(JUNIT_REPORT) > $(JUNIT_OUT))
 endif
-- 
2.35.3

